xsd-core/src/main/java/io/ballerina/xsd/core/context/XSDContext.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.xsd.core.context;
+
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.xsd.core.node.Kind;
+import io.ballerina.xsd.core.node.MemberNode;
+import io.ballerina.xsd.core.node.XSDElement;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents the context for processing XSD elements.
+ * This class provides methods to manage and retrieve syntax tree nodes and XSD elements.
+ */
+public final class XSDContext implements Context {
```

Comment:
Do we need this as public?

Reply:
Fixed in [f1a6ca7](https://github.com/ballerina-platform/xsd-tools/pull/26/commits/f1a6ca7780c7fe6953ec34c2533585e1c22fbc69)